virtual_rainforest/models/soil/soil_model.py

```diff
@@ -98,6 +98,15 @@ def __init__(
         """A Data instance providing access to the shared simulation data."""
         self.update_interval
         """The time interval between model updates."""
+        # Find first soil layer from the soil temperature in data object
+        self.top_soil_layer_index = next(
+            i
+            for i, v in enumerate(data["soil_temperature"].coords["layer_roles"])
+            if v == "soil"
```

Comment:
Better take index from `soil_moisture` which will definitely be initialized in the `setup()`